lib/spack/llnl/util/filesystem.py

```diff
@@ -2236,30 +2236,27 @@ def link_dest(self):
         if hasattr(self.pkg, "libs") and self.pkg.libs:
             pkg_libs = set(self.pkg.libs.directories)
         else:
-            pkg_libs = set((self.pkg.prefix.lib, self.pkg.prefix.lib64))
+            pkg_libs = set((self.pkg.prefix.lib, self.pkg.prefix.lib64, self.pkg.prefix.bin))
+        return pkg_libs
 
-        return pkg_libs | set([self.pkg.prefix.bin]) | self.internal_links
-
-    @property
-    def internal_links(self):
-        """
-        linking that would need to be established within the package itself. Useful for links
-        against extension modules/build time executables/internal linkage
-        """
-        return self._internal_links
-
-    def add_internal_links(self, *dest):
+    def add_library_dependent(self, *dest):
         """
-        Incorporate additional paths into the rpath (sym)linking scheme.
-
-        Paths provided to this method are linked against by a package's libraries
-        and libraries found at these paths are linked against a package's binaries.
-        (i.e. /site-packages -> /bin and /bin -> /site-packages)
+        Add paths to set of common paths that need to link against other libraries
 
-        Specified paths should be outside of a package's lib, lib64, and bin
+        Specified paths should be outside of a package's common link paths, i.e. the lib, lib64, and bin
         directories.
         """
-        self._internal_links = self._internal_links | set(*dest)
+        self._internal_links = self._internal_links | set(dest)
```

Comment:
If we're updating the names of the methods, I think the properties could also have clearer names:

* `self._internal_links`
* `.internal_link_targets`
* `link_dest`: ambiguity: is it "the places where symlinks should be pointing" or "the places where symlinks should be placed"? I think it's the latter from reading the code, but I mean to illustrate the ambiguity; I think being consistent with "dependent" as a term would be useful

Overall I think it would be good to constrain the terminology to "dependent" and "search_path"/"rpath"/"target" (any one of those 3 is fine, but don't use more than one)